### PR TITLE
LPS-196452 Use GetObjectFieldTypesMVCRender to Receive Localizable Business Types

### PR DESCRIPTION
**Pull request by igor-franca**
related [ticket](https://liferay.atlassian.net/browse/LPS-196452)

### CODE REVIEW

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/Fields.tsx [hunk 1]
```diff
@@ -51,6 +53,12 @@ export default function Fields({
 		setDeletedObjectField,
 	] = useState<ObjectField | null>(null);
 
+	const [load, setLoad] = useState<boolean>(false);
+
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/Fields.tsx:57]: Hey @igor-franca, can you rename this state to `loading` and `setLoading`?

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/Fields.tsx [hunk 2]
```diff
@@ -51,6 +53,12 @@ export default function Fields({
...
+	const [objectFieldTypes, setObjectFieldTypes] = useState<ObjectFieldType[]>(
+		[]
+	);
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/Fields.tsx:60]: Hey @igor-franca, you can create this useState directlly as a map: 
```suggestion
	const [objectFieldTypes, setObjectFieldTypes] = useState<
		Map<string, ObjectFieldType>
	>();
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/Fields.tsx [hunk 3]
```diff
@@ -68,16 +76,54 @@ export default function Fields({
...
 
 		makeFetch();
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [objectDefinitionExternalReferenceCode]);
 
+	function objectFieldBusinessTypeDataRenderer({
+		itemData,
+	}: {
+		itemData: ItemData;
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/Fields.tsx:116]: @igor-franca ItemData is actually ObjectField, we can create a new interface to replace all usage of this ItemData interface

```tsx
interface ObjectFieldItemData {
	itemData: ObjectField;
}
```
And replace all other functions that use it with something like:

```suggestion
: ObjectFieldItemData) 
```

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/Fields.tsx [hunk 4]
```diff
@@ -68,16 +76,54 @@ export default function Fields({
...
+	}) {
+		const businessTypeMap = new Map<string, ObjectFieldType>();
+
+		objectFieldTypes.forEach((objectFieldType) => {
+			businessTypeMap.set(objectFieldType.businessType, objectFieldType);
+		});
```
Comment (danvb112) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/Fields.tsx:122]: Hey @igor-franca, in this case, for each field we will create a `map`, do a `forEach` to define the map and `get` an item in a map. We can optimize this entire process by doing it once, in the `useEffect` above and configuring `setObjectFieldTypes` with this map. This `objectFieldBusinessTypeDataRenderer` function will only be used to obtain the value;

```tsx
function objectFieldBusinessTypeDataRenderer({
   itemData,
}: ObjectFieldItemData) {
   return objectFieldTypes?.get(itemData.businessType)?.label;
}
```

**Review by danvb112 — CHANGES_REQUESTED** [review]
Hey @igor-franca, Good job on this, I have some suggestions, fell free to ask :smile:

**Comment by igor-franca** [comment]
Thanks for suggesting these changes @danvb112! I took the liberty of renaming the state and creating the `objectFieldBusinessTypesMap` inline

**Review by danvb112 — APPROVED** [review]
LGTM! :tada:

**Comment by danvb112** [comment]
Hey @igor-franca, this PR is free to go now! Just resolve the conflicts and run the format command to see if everything is ok

**Comment by igor-franca** [comment]
ci:test:relevant

**Comment by igor-franca** [comment]
ci:test:sf

**Comment by andreefarias** [comment]
ci:test:object

**Comment by andreefarias** [comment]
Hey @igor-franca,

I noticed that this PR is generating a new critical bug. The fields tab from the object definition that receive a **relationship field** shows a blank page.

[Testcases related](https://testray.liferay.com/home/-/testray/subtasks/2462108356)

See: 
image

**Pull request closed without merging**